packages/commercetools/composables/src/getters/cartGetters.ts

```diff
@@ -48,7 +48,18 @@ export const getCartTotals = (cart: Cart): AgnosticTotals => {
   };
 };
 
-export const getCartShippingPrice = (cart: Cart): number => cart && cart.shippingInfo ? cart.shippingInfo.price.centAmount / 100 : 0;
+export const getCartShippingPrice = (cart: Cart): number => {
+  if (cart?.shippingInfo) {
+    if (cart.shippingInfo?.shippingMethod?.zoneRates[0].shippingRates[0].freeAbove) {
+      const total = cart.totalPrice.centAmount;
+      if (total >= cart.shippingInfo?.shippingMethod?.zoneRates[0].shippingRates[0].freeAbove.centAmount) {
+        return 0;
+      }
+    }
+    return cart?.shippingInfo ? cart.shippingInfo.price.centAmount / 100 : 0;
+  }
+  return 0;
+};
```

Comment:
I think it's hard to read the code when we have many nested `if` statements. The code above is fine, but I suggest flattening it a bit:

```javascript
export const getCartShippingPrice = (cart: Cart): number => {
  const total = cart?.totalPrice?.centAmount;
  const shippingInfo = cart?.shippingInfo;
  const centAmount = shippingInfo?.shippingMethod?.zoneRates[0].shippingRates[0].freeAbove?.centAmount;

  if (!shippingInfo || !centAmount || !total || total >= centAmount) {
    return 0;
  }

  return shippingInfo.price.centAmount;
};
```

I haven't tested this code. Please do it if you decide to use it.